parso/python/tree.py

```diff
@@ -549,7 +548,15 @@ class Function(ClassOrFunc):
     def __init__(self, children):
         super().__init__(children)
         parameters = self.children[2]  # After `def foo`
-        parameters.children[1:-1] = _create_params(parameters, parameters.children[1:-1])
+        parameters_children = parameters.children[1:-1]
+        input_has_param = False
+        for child in parameters_children:
+            if isinstance(child, Param):
+                input_has_param = True
+        # If input parameters list already has Param objects, keep it as is;
+        # otherwise, convert it to a list of Param objects.
+        if not input_has_param:
+            parameters.children[1:-1] = _create_params(parameters, parameters_children)
```

Comment:
Can we use something like this here instead of an expanded for loop with variable overwrites (same goes below)?
`not any(isinstance(child, Param) for child in parameters.children[1:-1])`

Reply:
Yes we can definitely do that. The old version of this PR has more logic under the `if` branch and cannot be simplified into an `any` call with comprehension. Later the logic was removed but I forgot to simplify the whole piece of code.